### PR TITLE
feat: add `db.ignore_unknown_table_columns` configuration (#3192)

### PR DESCRIPTION
**Pull request by mih-kopylov**
The property allows to ignore scan errors when columns in the SQL result have no fields in the destination struct.

## Related issue(s)

#3192 - it's described as a bug, and in order to fix the bug a configuration option is introduced, so that the existing behaviour is kept by default, while a new mode could be enabled.

I'm also happy for the new mode to be applied by default without a configuration flag, just let me know if it's the case.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I am following the [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

**Comment by mih-kopylov** [comment]
I don't have any idea why some of the pipelines failed, especially one that validate JS or dockle.

It might be helpful to reflect the change in documentation, and an advise with a specific docs section to be updated is highly appreciated

File: driver/registry_sql.go [hunk 1]
```diff
@@ -86,6 +86,7 @@ func (m *RegistrySQL) Init(ctx context.Context) error {
 			Pool:                      pool,
 			UseInstrumentedDriver:     m.Tracer(ctx).IsLoaded(),
 			InstrumentedDriverOptions: opts,
+			Unsafe:                    m.C.DbIgnoreUnknownTableColumns(),
```
Comment (aeneasr) [inline driver/registry_sql.go:89]: Wow, I didn't know the solution is that easy! I thought it needed much more work. Great stuff! :)

**Review by aeneasr — CHANGES_REQUESTED** [review]
Thank you! :) Can you please add a test for provider.go to ensure that the setting works correctly? You will also need to add the config value to `spec/config.json` :)

Otherwise LGTM

**Comment by mih-kopylov** [comment]
@aeneasr both done.
The only thing - it needs to have a real connection to a database for a proper test, so I used a postgres in docker and run the same query with both modes. Hope that's fine.

Can I kindly ask this patch to be a part of the next 1.11.x please?

**Comment by mih-kopylov** [comment]
@aeneasr just a kind reminder

**Comment by aeneasr** [comment]
Sorry, I dropped the ball on this one. The CI is currently failing, can you please take a look?

**Comment by aeneasr** [comment]
@mih-kopylov the tests added in the PR seem to not be working: https://github.com/ory/hydra/runs/8015017590?check_suite_focus=true#step:12:40

**Comment by mih-kopylov** [comment]
Thanks, I'll fix that once able to checkout `master` on Windows. Kinda chain of issues :)

**Comment by aeneasr** [comment]
Ok, we'll get that one merged asap also

**Comment by aeneasr** [comment]
i merged the fix to master

**Review by aeneasr — APPROVED** [review]
Awesome, thank you! 🎉 Your contribution makes Ory better :)

**Pull request merged**